### PR TITLE
Examples- new pr

### PR DESCRIPTION
**Pull request by TomasRoj**
Okay, updated something. If it will not work lets ignore it. I am not good at this. Maybe it would be best dont do anything...

### CODE REVIEW

**Comment by TomasRoj** [comment]
Only error I see that it doesn't recognize piezo1 from you library. Is it all capital letters?

**Comment by JarekParal** [comment]
Please read the [documentation](https://github.com/RoboticsBrno/ArduinoLearningKitStarter-library#piezo). Yes, it must be all capital letters. You can also see that in [source code](https://github.com/RoboticsBrno/ArduinoLearningKitStarter-library/blob/master/src/ALKS.h).

**Comment by TomasRoj** [comment]
So if it will be capital letters will be everything okay?

**Comment by TomasRoj** [comment]
I mean whole pull request

**Comment by TomasRoj** [comment]
Updated now. It still doesn't work 😔

**Comment by bsaid** [comment]
> Updated now. It still doesn't work

What exactly does not work? I have successfully compiled "buzzerSongs_Pirates" code for Arduino Nano and Uno. I actually don't have real hardware to test it.

**Comment by JarekParal** [comment]
[The Travis.ci failed](https://travis-ci.org/RoboticsBrno/ArduinoLearningKitStarter-library/builds/520441023#L647). This is probably a problem with ESP32 and missing function tone().

**Comment by bsaid** [comment]
> [The Travis.ci failed](https://travis-ci.org/RoboticsBrno/ArduinoLearningKitStarter-library/builds/520441023#L647). This is probably a problem with ESP32 and missing function tone().

I see, I suppose that @TomasRoj creates the examples on Arduino, not on ESP32. Am I true? If something is not working on Arduino, I would like to know about it. We can solve the ESP32 incompatibility later.

**Comment by JarekParal** [comment]
Ok, we can have some examples just for Uno/Nano. But then somebody must solve this exeption in Travis.ci, because we want/need the CI.

Or there is another way with `#ifdef ESP32`  and use the `ledcWriteTone()` function which has ESP32: https://techtutorialsx.com/2017/07/01/esp32-arduino-controlling-a-buzzer-with-pwm/

Or there is also this issue: https://github.com/RoboticsBrno/ToneAC/issues/1

**Comment by TomasRoj** [comment]
Okay. So everything on my side is OK?

**Comment by TomasRoj** [comment]
I can write what @JarekParal said

**Comment by TomasRoj** [comment]
Guys Problem is in Travis. I tried **Normal** code for uno/nano and it works on my wemos d1 too. there must be problem with travis ci tests.

**Comment by TomasRoj** [comment]
guys, whole travis ci test is wrong. Even the only one example (pwmLed) what I left doesnt work on esp. It works on my board at home.

**Comment by JarekParal** [comment]
No, the problem is with ESP32 which hasn't got support for standard function `analogWrite()` (which is available on Uno/Nano and probably also on ESP8266). This is the same problem as with the library ToneAc. Please read more about that and try to find some solution.

Posible sources of informations:
- https://techtutorialsx.com/2017/07/01/esp32-arduino-controlling-a-buzzer-with-pwm/
- https://github.com/espressif/arduino-esp32/issues/4
- https://randomnerdtutorials.com/esp32-pwm-arduino-ide/

**Comment by TomasRoj** [comment]
I already tried it. But when I did exception for esp travis ignored.

**Comment by TomasRoj** [comment]
I dont know.. Why it can be only on uno/nano?

**Comment by TomasRoj** [comment]
Still same problem even after I changed thngs you said: /home/travis/build/RoboticsBrno/ArduinoLearningKitStarter-library/examples/pwmLed/pwmLed.ino:1:2: warning: #import is a deprecated GCC extension [-Wdeprecated]
 #import <ALKS.h>
  ^
pwmLed:15: error: conflicting declaration 'const int ledPin'
 const int ledPin = 16;  // 16 corresponds to GPIO16
           ^
/home/travis/build/RoboticsBrno/ArduinoLearningKitStarter-library/examples/pwmLed/pwmLed.ino:6:5: note: previous declaration as 'int ledPin'
 int ledPin = 8;     
     ^
/home/travis/build/RoboticsBrno/ArduinoLearningKitStarter-library/examples/pwmLed/pwmLed.ino: In function 'void setup()':
pwmLed:18: error: 'ledcSetup' was not declared in this scope
   ledcSetup(ledChannel, freq, resolution);
                                         ^
pwmLed:21: error: 'ledcAttachPin' was not declared in this scope
   ledcAttachPin(ledPin, ledChannel);

**Comment by JarekParal** [comment]
Why you don't try firstly compilation of the code locally on your PC with setting for Una/Nano and then for ESP32 ([how to add esp32 to Arduino IDE](https://github.com/espressif/arduino-esp32/blob/master/docs/arduino-ide/boards_manager.md)) and fix all the error which you get? Or you can one again read the message from travis and fix the bugs:

> /home/travis/build/RoboticsBrno/ArduinoLearningKitStarter-library/examples/pwmLed/pwmLed.ino:1:2: warning: #import is a deprecated GCC extension [-Wdeprecated]
#import <ALKS.h>
^
pwmLed:15: error: conflicting declaration 'const int ledPin'
const int ledPin = 16; // 16 corresponds to GPIO16
^
/home/travis/build/RoboticsBrno/ArduinoLearningKitStarter-library/examples/pwmLed/pwmLed.ino:6:5: note: previous declaration as 'int ledPin'
int ledPin = 8;

+ one hint: 
ledcSetup() and ledcAttachPin() are function available only for ESP32

**Comment by TomasRoj** [comment]
I tought we are developing together to help beginners with arduino not playing az-quiz

**Comment by JarekParal** [comment]
No, we are trying to help you learn new thing and became a better programmer ;).

**Comment by TomasRoj** [comment]
IDK. It shows me same bug: error: 'ledcSetup' was not declared in this scope. Am i missing some library or something?

**Comment by JarekParal** [comment]
Let's think. For which MCU you build the project? Which MCU has available `ledcSetup()` function?

**Comment by TomasRoj** [comment]
for esp32

**Comment by JarekParal** [comment]
Which ESP32 board you select in the Arduino IDE?

File: examples/pwmLed/pwmLed.ino [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+int ledPin = 8;     
+
+int potProm = 0;    
+int ledProm = 0;   
+
+const int freq = 5000; 
+const int ledChannel = 0; 
+const int resolution = 8;
+
+const int ledPin = 16;  // 16 corresponds to GPIO16
```
Comment (cleveHEX) [inline examples/pwmLed/pwmLed.ino]: Duplicate definition of existing variable can't be `const`

File: examples/pwmLed/pwmLed.ino [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#import <ALKS.h>
+
+//We define all pins
+
+int potPin = A0;   
+int ledPin = 8;
```
Comment (cleveHEX) [inline examples/pwmLed/pwmLed.ino:6]: This

**Review by cleveHEX — CHANGES_REQUESTED** [review]
Look at this...

**Comment by JarekParal** [comment]
Here is advice, how to fix the problem with ESP32 compilation.
```c
  #if defined(ESP32)
    ledcSetup(ledChannel, freq, resolution);
    
    // attach the channel to the GPIO to be controlled
    ledcAttachPin(ledPin, ledChannel);
  #endif
```

**Comment by cleveHEX** [comment]
> 
> 
> Still same problem even after I changed thngs you said: /home/travis/build/RoboticsBrno/ArduinoLearningKitStarter-library/examples/pwmLed/pwmLed.ino:1:2: warning: #import is a deprecated GCC extension [-Wdeprecated]
> #import <ALKS.h>
> ^
> pwmLed:15: error: conflicting declaration 'const int ledPin'
> const int ledPin = 16; // 16 corresponds to GPIO16
> ^
> /home/travis/build/RoboticsBrno/ArduinoLearningKitStarter-library/examples/pwmLed/pwmLed.ino:6:5: note: previous declaration as 'int ledPin'
> int ledPin = 8;
> ^
> /home/travis/build/RoboticsBrno/ArduinoLearningKitStarter-library/examples/pwmLed/pwmLed.ino: In function 'void setup()':
> pwmLed:18: error: 'ledcSetup' was not declared in this scope
> ledcSetup(ledChannel, freq, resolution);
> ^
> pwmLed:21: error: 'ledcAttachPin' was not declared in this scope
> ledcAttachPin(ledPin, ledChannel);

Read my comments on #16

**Comment by TomasRoj** [comment]
Thanks for help, I will look at that in the afternoon

**Comment by TomasRoj** [comment]
@JarekParal seems like your code is wrong. I tried it and travis shows me error that #endif cannot be used without #if even it is there,

**Comment by bsaid** [comment]
@TomasRoj I have reviewed your examples and modified them to work on both ESP32 and Arduino. I tested the modified code only on ESP32. If you are satisfied with the modification, you can commit it to your code and your pull request will be probably accepted.
I know, there is a complication with the code incompatibility between ESP32 and Atmel based Arduino. We will maybe separate examples for ESP32 and for Arduino. There are other reasons to do so, for example, WiFi on the ESP32.
https://github.com/RoboticsBrno/ArduinoLearningKitStarter-library/tree/ExamplesTestBasedOnTomasRoj/examples

**Comment by TomasRoj** [comment]
Now I added files you created. It should work now

**Comment by TomasRoj** [comment]
So no it doesn't work still. It says that analog write function was not declared...

**Comment by cleveHEX** [comment]
You seem like you have no idea what are you actually doing, the other one is failing build, the first one is not even building, because it was not named `*.ino`.

**Comment by TomasRoj** [comment]
No it has that same problem with analog write function

**Comment by cleveHEX** [comment]
> 
> 
> No it has that same problem with analog write function

image
it passed build on esp32, I will have to test it on real HW, though.

**Comment by TomasRoj** [comment]
Yes but no PWM Led example

**Comment by cleveHEX** [comment]
@JarekParal 
someone made temporary esp32 tone library, you can check it [here](https://github.com/lbernstone/Tone).

**Comment by TomasRoj** [comment]
@bsaid said that it's working on real esp32 both examples. We can merge

**Comment by cleveHEX** [comment]
> 
> 
> @bsaid said that it's working on real esp32 both examples. We can merge

No:
- It does not build, thus it can't work on esp32.
- It does not meet code quality standard at the moment.

**Comment by TomasRoj** [comment]
Then look at what he wrote and consult it with him I did maximum

**Comment by bsaid** [comment]
@TomasRoj The CI does not work, because in file pwmLed.ino at line 38 is "#endif", but should be "#else". The code below line 38 can run only on Arduino, but with wrong "#endif" line it run on both platforms.

**Comment by cleveHEX** [comment]
Anyways, I don't like such things being in examples, it is too hard to understand for ALKS end users (people with no knowledge). Examples should be easy to understand for anyone, without complex blobs. I will may implement HAL of `analogWrite` and `tone` soon to library (#20, #21).

**Comment by bsaid** [comment]
> Anyways, I don't like such things being in examples, it is too hard to understand for ALKS end users (people with no knowledge). Examples should be easy to understand for anyone, without complex blobs. I will may implement HAL of `analogWrite` and `tone` soon to library (#20, #21).

@CornyjK I fully agree with you. But please, solve this PR first, and then we can improve the examples to be easier for beginners.

**Comment by cleveHEX** [comment]
It's still unclear where have you stolen the code, I can't merge any code without proper credits anyways.

**Comment by bsaid** [comment]
I have updated the pwmLed.ino in my test branch: https://github.com/RoboticsBrno/ArduinoLearningKitStarter-library/tree/ExamplesTestBasedOnTomasRoj/examples/pwmLed

Sorry, I pushed the changes previously only for Pirates example.

**Comment by bsaid** [comment]
@TomasRoj So, now my commit 6e152da91cef53987bab45438c515ac9f29d62e8 is accepted by the CI. Sorry for this.

**Comment by TomasRoj** [comment]
So finsihed pr?

**Comment by cleveHEX** [comment]
> 
> 
> So finsihed pr?

You did not met all requirements which I stated in this or #16 PR.

**Comment by cleveHEX** [comment]
> 
> 
> @TomasRoj So, now my commit [6e152da](https://github.com/RoboticsBrno/ArduinoLearningKitStarter-library/commit/6e152da91cef53987bab45438c515ac9f29d62e8) is accepted by the CI. Sorry for this.

You can directly edit his code, so he does not need to recommit your changes.

**Comment by cleveHEX** [comment]
The LED PWM is already done in #23, so it is not needed anyways.

**Comment by TomasRoj** [comment]
Maybe, in the time when we will discuss this pr I can help you with documentation or something similar to that. What do you think? Is something in work-in-progress ?

**Pull request closed without merging**